modules/internal/ChapelArray.chpl

```diff
@@ -1815,17 +1823,33 @@ module ChapelArray {
             return true;
           }
         }
+        // We didn't find it, so return false.
+        return false;
+
+      // workaround for #21749:
+      } else if __primitive("call and fn resolves", "max", fullIdxType) {
```

Comment:
I am OK with this, but it is somewhat arbitrary, right?

If I have an associative domain of strings, I'll fall back to serial case. Which also makes me wonder whether that'll truly set `idx` to the "lexicographically earliest index" as the doc suggests. (I would think the iteration order of associative domain of strings is undefined?)

In any case, I realize that these are more tangential thoughts than being helpful for this PR. Even if there's a limitation, this is definitely a step forward.

Reply:
> Which also makes me wonder whether that'll truly set idx to the "lexicographically earliest index" as the doc suggests. (I would think the iteration order of associative domain of strings is undefined?)

Oh, that's a great point...  and a pre-existing bug that slipped past me and my first reviewer (since the serial version is the same as before).  Let me try to add a test for that as well as a fix.

Reply:
https://github.com/chapel-lang/chapel/pull/21750/commits/a376ba44de4ea587b1105afbbd644b944e7093f9 should fix this problem albeit in a naive way, if you'd give it a look.

Reply:
Changes in that commit looks good and make sense.